src/value.c

```diff
@@ -259,6 +273,41 @@ const char *SIType_ToString(SIType t) {
 	}
 }
 
+void SIType_ToMultipleTypeString(SIType t, char **buf, size_t *bufferLen, size_t *bytesWritten) {
+	// Worst case: Len(SIType names) + 19*Len(", ") + Len("Or") = 177 + 38 + 2 = 217
+	ASSERT((*bufferLen - *bytesWritten) >= 256);
+
+	SIType currentType;
+	SIType remainingTypes = t;
+	
+	uint typesDetected = 0;
+	// Iterate over the possible SITypes
+	for(currentType = 1; currentType < SI_ALL; currentType = currentType << 1) {
```

Comment:
Please look into `popcount` which should return the number of active bits in an integer.

Reply:
__builtin_popcount() is used now. The for() was replaced by while()